R/SIRD_plot.R

```diff
@@ -1,121 +1,158 @@
+        func = equationsSIRD,
         parms = c(
-          beta = input$betaSIRD,
-          gamma = input$gammaSIRD,
-          delta = input$deltaSIRD,
-          muB = input$muBirth,
-          muD = input$muDeath
+            beta = beta,
+            gamma = gamma,
+            delta = delta,
+            muB = muB,
+            muD = muD
         )
-      )
-    })
+    )
 
-    output$plotSIRD <- renderPlot({
-      input$go
-      isolate({
-        val <- as.data.frame(sird_values())
-        ggplot(val, aes(x = time)) +
-          theme(
-            axis.line = element_line(color = "black"), axis.text = element_text(size = 14),
+    as.data.frame(solnsSIRD)
+}
+
+plotSIRD <- function() {
+    valuesSIRD <- solveSIRD()
+    ggplot(valuesSIRD, aes(x = time)) +
+        theme(
+            axis.line = element_line(color = "black"),
+            axis.text = element_text(size = 14),
             axis.title.x = element_text(size = 16, face = "bold"),
             axis.title.y = element_text(size = 16, face = "bold")
-          ) +
-          ggtitle("SIRD Epidemic Model") +
-          theme(plot.title = element_text(size = 22, face = "bold")) +
-          theme(legend.position = "bottom") +
-          ylab("Number of People") +
-          xlab("Time") +
-          scale_x_continuous(expand = c(0, 0)) +
-          scale_y_continuous(expand = c(0, 0)) +
-          geom_line(aes(y = S, color = "Blue"), linewidth = 1.5) +
-          geom_line(aes(y = I, color = "Red"), linewidth = 1.5) +
-          geom_line(aes(y = R, color = "Green"), linewidth = 1.5) +
-          geom_line(aes(y = D, color = "Orange"), linewidth = 1.5) +
-          scale_color_identity(
+        ) +
+        ggtitle("SIRD Epidemic Model") +
+        theme(plot.title = element_text(size = 22, face = "bold")) +
+        theme(legend.position = "bottom") +
+        ylab("Number of People") +
+        xlab("Time") +
+        scale_x_continuous(expand = c(0, 0)) +
+        scale_y_continuous(expand = c(0, 0)) +
+        geom_line(aes(y = S, color = "Blue"), linewidth = 1.5) +
+        geom_line(aes(y = I, color = "Red"), linewidth = 1.5) +
+        geom_line(aes(y = R, color = "Green"), linewidth = 1.5) +
+        geom_line(aes(y = D, color = "Orange"), linewidth = 1.5) +
+        scale_color_identity(
             name = "SIRD", breaks = c("Blue", "Red", "Green", "Orange"),
             labels = c("Susceptible", "Infected", "Recovered", "Dead"), guide = "legend"
-          )
-      })
-    })
+        )
+}
+
 
-    output$SIRDPhasePlane <- renderPlot({
-      input$go
-      isolate({
-        val <- as.data.frame(sird_values())
-        ggplot(val, aes(x = S)) +
-          geom_line(aes(y = I, color = "Blue"), linewidth = 1.5) +
-          theme(
-            axis.line = element_line(color = "black"), axis.text = element_text(size = 14),
+plotPhasePlaneSIRD <- function() {
+    valuesSIRD <- solveSIRD()
+    ggplot(valuesSIRD, aes(x = S)) +
```

Comment:
Please replace `valuesSIRD` here with the function call itself. The assignment is unnecessary.